pennylane/interfaces/autograd.py

```diff
@@ -195,18 +195,81 @@ def vjp(ans, self, params, device):  # pylint: disable=unused-argument
             gradient output vector, and computes the vector-Jacobian product
         """
 
-        def gradient_product(g):
-            # In autograd, the forward pass is always performed prior to the backwards
-            # pass, so we do not need to re-unwrap the parameters.
+        saved_grad_matrices = {}

```

Comment:
Reading over other people's comments, we should really have a comment about closure here.  I can see someone trying to modify this later on, breaking things, and being confused. 

Maybe something along the lines of 
```
# python stores this with `_evaluate_grad_matrix` and cleans them up together
# mutable data type can be modified inside the inner function
```

Maybe a link to an explanation like [this one that helped me](https://realpython.com/inner-functions-what-are-they-good-for/#retaining-state-in-a-closure).

Reply:
Very good idea, I'll do that now.

Reply:
Done in 4c825cd